Sources/SpeziOnboarding/ConsentView/ConsentView.swift

```diff
@@ -0,0 +1,405 @@
+                case .error(let error):
+                    viewState = .base(.error(error))
+                }
+            }
+            .sheet(isPresented: $showShareSheet) {
+                switch viewState {
+                case .exported(let consent):
+                    ShareSheet(sharedItem: consent)
+                        .presentationDetents([.medium])
+                default:
+                    EmptyView()
+                }
+            }
+            .viewStateAlert(state: contentViewStateBinding)
+        }
+    }
+    
+    private var inputFieldsDisabled: Bool {
+        switch viewState {
+        case .base(.processing), .export(_): true  // swiftlint:disable:this empty_enum_arguments
+        default: false
+        }
+    }
+    
+    private var actionButtonsDisabled: Bool {
+        let showSignatureView = !(name.givenName?.isEmpty ?? true) && !(name.familyName?.isEmpty ?? true)
+        if !self.showSignatureView && showSignatureView {
+            Task { @MainActor in
+                self.showSignatureView = showSignatureView
+            }
+        }
+        
+        if signature.strokes.isEmpty || (name.givenName?.isEmpty ?? true) || (name.familyName?.isEmpty ?? true) {
+            return true
+        } else {
+            // As soon as form is complete, set it to signed
+            Task { @MainActor in
+                if viewState == .base(.idle) {
+                    viewState = .signed
+                }
+            }
+            return false
+        }
+    }
+    
+    
+    /// Creates a ``ConsentView`` with a provided action view using  an``OnboardingActionsView`` and renders a markdown view.
+    /// Furthermore, by default, the signed consent form is exported to the `Standard` as a PDF.
+    /// - Parameters:
+    ///   - header: The header view will be displayed above the markdown content.
+    ///   - asyncMarkdown: The markdown content provided as an UTF8 encoded `Data` instance that can be provided asynchronously.
+    ///   - footer: The footer view will be displayed above the markdown content.
+    ///   - action: The action that should be performed once the consent has been given.
+    ///   - givenNameField: The localization to use for the given (first) name field.
+    ///   - familyNameField: The localization to use for the family (last) name field.
+    ///   - exportConsentForm: Indicates weather the signed consent form should be exported as a PDF to the `Standard`. Defaults to true.
+    public init(
+        @ViewBuilder header: () -> some View = { EmptyView() },
+        asyncMarkdown: @escaping () async -> Data,
+        @ViewBuilder footer: () -> some View = { EmptyView() },
+        givenNameField: FieldLocalizationResource = LocalizationDefaults.givenName,
+        familyNameField: FieldLocalizationResource = LocalizationDefaults.familyName,
+        exportConfiguration: ExportConfiguration = .init(),
+        action: @escaping () async -> Void
+    ) {
```

Comment:
I have taken a look at the PR and as noted in the comment on the main Consent file, I would suggest to even more drastically simplify the public UI and allow users to use the consent view in a more custom UI that does not include the `OnboardingActionButtons`, we already have one application that would have this as a clear use case.

Exposing the public API like this would allow you to display the `ConsentDocument` view within an arbitrary view hierarchy and or use the more sophisticated and all-batteries included `ConsentOnboardingView` in a typical onboarding setting like the Template Application.

Removing the header and footer views also makes it more clearer that the consent document only contains the document, signature, and everything displayed by this view and under control of this view instead of arbitrarily added header and footer views.
```suggestion
    public init(
        markdown: @escaping () async -> Data,
        givenNameField: FieldLocalizationResource = LocalizationDefaults.givenName,
        familyNameField: FieldLocalizationResource = LocalizationDefaults.familyName,
        exportConfiguration: ExportConfiguration = .init(),
        viewState: ViewState
    ) {
```

In the long run: We have seen that these header and footer views in the Onboarding have caused several issues and were hard to understand. We might move away from then on the long run and rather ask developers to take advantage of the SwiftUI composability and e.g. just use a `VStack` themselves.